packages/venia-ui/lib/components/SearchPage/searchPage.js

```diff
@@ -129,25 +134,32 @@ const SearchPage = props => {
 
     return (
         <article className={classes.root}>
-            <div className={classes.categoryTop}>
-                <span className={classes.totalPages}>
-                    {formatMessage(
-                        {
-                            id: 'searchPage.totalPages',
-                            defaultMessage: `items`
-                        },
-                        { totalCount: productsCount }
-                    )}
-                </span>
-                <div className={classes.headerButtons}>
-                    {maybeFilterButtons}
-                    {maybeSortButton}
+            <div className={classes.sidebar}>
+                <Suspense fallback={null}>{maybeSidebar}</Suspense>
+            </div>
+            <div className={classes.searchContent}>
+                <div className={classes.heading}>
+                    <div className={classes.searchInfo}>
+                        {searchResultsHeading}
+                        <span className={classes.totalPages}>
+                            {formatMessage(
+                                {
+                                    id: 'searchPage.totalPages',
+                                    defaultMessage: `items`
+                                },
+                                { totalCount: productsCount }
+                            )}
```

Comment:
If there are search results and then you apply a filter, this count flashes `0 items` for a moment.

I know you didn't add this code, but could you experiment with making the rendering of this text conditional on the results being loaded? Perhaps the `loading` prop would work.

[![Image from Gyazo](https://i.gyazo.com/6953d781162b2bec0760206078acde88.gif)](https://gyazo.com/6953d781162b2bec0760206078acde88)

Reply:
@sirugh I did actually introduce this issue. It's now fixed!